src/typescript/typescript.ts

```diff
@@ -22,6 +23,158 @@ import {
 } from "../typescript";
 import { deepMerge } from "../util";
 
+export class TsJestBabelConfig {
+  public static default() {
+    return new TsJestBabelConfig(true);
+  }
+
+  public static fromFile(filePath: string) {
+    return new TsJestBabelConfig(filePath);
+  }
+
+  public static custom(config: Record<string, any>) {
+    return new TsJestBabelConfig(config);
+  }
+
+  private constructor(
+    private readonly config: boolean | string | Record<string, any>
+  ) {}
+
+  /**
+   * @jsii ignore
+   * @internal
+   */
+  public toJSON(): boolean | string | Record<string, any> {
+    return this.config;
+  }
+}
+
+export class TsJestDiagnostics {
+  public static default() {
+    return new TsJestDiagnostics(true);
+  }
+
+  public static disabled() {
+    return new TsJestDiagnostics(false);
+  }
+
+  public static custom(config: Record<string, any>) {
+    return new TsJestDiagnostics(config);
+  }
+
+  private constructor(private readonly config: boolean | Record<string, any>) {}
+
+  /**
+   * @jsii ignore
+   * @internal
+   */
+  public toJSON(): boolean | Record<string, any> {
+    return this.config;
+  }
+}
+
+export class TsJestTsconfig {
+  public static default() {
+    return new TsJestTsconfig(true);
+  }
+
+  public static disabled() {
+    return new TsJestTsconfig(false);
+  }
```

Comment:
Not sure about this name, but don't have a better one right now.

```suggestion
  /**
   * Force` ts-jest` to use its built-in defaults even if there is a `tsconfig.json` in your project.
   */
  public static builtInDefaults() {
    return new TsJestTsconfig(false);
  }
```